CesiumGS/cesium-webpack-example
# Webpack 4 example

This PR adds a separate example for Webpack version 4 alongside Webpack 5. Ideally most people should be on version 5 as soon as possible and on all new products. However, older projects or ones using older versions of frameworks like Nuxt2 and older Angular versions still rely on Webpack version 4

There are some challenges wrapped in getting this to work for Webpack 4 which mostly stem from using newer JS features that Webpack doesn't know how to support. 

- Requires `babel` for transformations like optional chaining (`.?`)
- Requires a new loader to let Webpack understand `import.meta` (`@open-wc/webpack-import-meta-loader`)
    - This seems not ideal and definitely seems to increase build time, may want to look into an alternative
- Requires ignoring "critical" errors about not being able to fully statically analyze cesium's code
    - Speaking with @ggetz this is unavoidable with the way cesium needs to be structured. However Webpack 5 doesn't require any extra config for this which is very curious to me

I also updated the github actions to make sure to perform the build checks and linting against both webpack 4 and 5 versions.

### Testing

- Change to both the `webpack-4` and `webpack-5` directories
- Run `npm install` 
- Run `npm run build`
- Run `npm start` and make sure the page works correctly at http://localhost:8080/
- Run `npm run start:built` and make sure the page works correctly at http://localhost:8080/

There should not be any errors in the console or on the page for both version 4 and 5
